### PR TITLE
feat: support debug mode for CN platform

### PR DESCRIPTION
**Pull request by hkbarton**
## What has been implemented?

support `--debug` for CN user.

### CODE REVIEW

File: src/cli/commands-cn/run.js [hunk 1]
```diff
@@ -60,6 +67,37 @@ module.exports = async (config, cli, command) => {
...
+      },
+      onEvent: (evt) => {
+        if (evt.event !== 'instance.run.logs') {
+          return;
+        }
+        if (evt.data.logs && Array.isArray(evt.data.logs)) {
+          evt.data.logs.forEach((log) => {
+            // Remove strange formatting that comes from stderr
+            if (typeof log.data === 'string' && log.data.startsWith("'")) {
+              log.data = log.data.substr(1);
```
Comment (medikoo) [inline src/cli/commands-cn/run.js:86]: `substr` is deprecated string method. Best is to just rely on `slice` (which shares same rules as `arr.slice` hence easy to remember)

File: src/cli/commands-cn/run.js [hunk 2]
```diff
@@ -60,6 +67,37 @@ module.exports = async (config, cli, command) => {
...
+        instanceName: instanceYaml.name,
+      },
+      onEvent: (evt) => {
+        if (evt.event !== 'instance.run.logs') {
+          return;
+        }
+        if (evt.data.logs && Array.isArray(evt.data.logs)) {
+          evt.data.logs.forEach((log) => {
+            // Remove strange formatting that comes from stderr
+            if (typeof log.data === 'string' && log.data.startsWith("'")) {
```
Comment (medikoo) [inline src/cli/commands-cn/run.js:85]: Can `log.data` be not of string type?

File: src/cli/commands-cn/run.js [hunk 3]
```diff
@@ -60,6 +67,37 @@ module.exports = async (config, cli, command) => {
...
+      filter: {
+        stageName: instanceYaml.stage,
+        appName: instanceYaml.app,
+        instanceName: instanceYaml.name,
+      },
+      onEvent: (evt) => {
+        if (evt.event !== 'instance.run.logs') {
+          return;
+        }
+        if (evt.data.logs && Array.isArray(evt.data.logs)) {
```
Comment (medikoo) [inline src/cli/commands-cn/run.js:82]: Could be just `if (Array.isArray(evt.data.logs)) {`

File: src/cli/commands-cn/runAll.js [hunk 4]
```diff
@@ -46,10 +53,38 @@ module.exports = async (config, cli, command) => {
...
+            }
+            if (typeof log.data === 'string' && log.data.endsWith('\\n')) {
+              log.data = log.data.substring(0, log.data.length - 2);
+            }
+            cli.log(log.data);
+          });
+        }
+      },
+    });
+  }
```
Comment (medikoo) [inline src/cli/commands-cn/runAll.js:86]: This looks as copy/pasted logic from `run`. Can we put it in utils, and reuse?

File: src/cli/commands-cn/run.js (hunk 1 above)
Comment (hkbarton) [reply src/cli/commands-cn/run.js:86]: done

File: src/cli/commands-cn/run.js (hunk 2 above)
Comment (hkbarton) [reply src/cli/commands-cn/run.js:85]: seems like it should be always string

File: src/cli/commands-cn/run.js (hunk 3 above)
Comment (hkbarton) [reply src/cli/commands-cn/run.js:82]: done

File: src/cli/commands-cn/runAll.js (hunk 4 above)
Comment (hkbarton) [reply src/cli/commands-cn/runAll.js:86]: done

File: src/cli/commands-cn/run.js [hunk 5]
```diff
@@ -116,5 +135,7 @@ module.exports = async (config, cli, command) => {
   cli.sessionStop('success', 'Success');
 
   if (deferredNotificationsData) printNotification(cli, await deferredNotificationsData);
+
+  sdk.disconnect();
```
Comment (medikoo) [inline src/cli/commands-cn/run.js:139]: What if connection was not initialized (with `options.debug` being falsy), wouldn't it crash?

File: src/cli/commands-cn/runAll.js [hunk 6]
```diff
@@ -95,4 +111,7 @@ module.exports = async (config, cli, command) => {
   cli.sessionStop('success', 'Success');
 
   if (deferredNotificationsData) printNotification(cli, await deferredNotificationsData);
+
+  sdk.disconnect();
```
Comment (medikoo) [inline src/cli/commands-cn/runAll.js:115]: Same here

File: src/cli/commands-cn/utils.js [hunk 7]
```diff
@@ -227,6 +227,29 @@ const setInputsForCommand = (instanceYaml, command, config) => {
...
+      evt.data.logs.forEach((log) => {
+        // Remove strange formatting that comes from stderr
+        if (log.data.startsWith("'")) {
+          log.data = log.data.slice(1);
+        }
+        if (log.data.endsWith("'")) {
+          log.data = log.data.substring(0, log.data.length - 1);
+        }
+        if (log.data.endsWith('\\n')) {
+          log.data = log.data.substring(0, log.data.length - 2);
```
Comment (medikoo) [inline src/cli/commands-cn/utils.js:245]: nit: for consistency it'll be nice to rely on `slice` here as well (it'll even be simpler)

File: src/cli/commands-cn/run.js (hunk 5 above)
Comment (hkbarton) [reply src/cli/commands-cn/run.js:139]: it won't, disconnect is safe: https://github.com/serverlessinc/platform-cn/blob/master/sdk/src/Connection.js#L81

File: src/cli/commands-cn/utils.js (hunk 7 above)
Comment (hkbarton) [reply src/cli/commands-cn/utils.js:245]: done

File: src/cli/commands-cn/utils.js [hunk 8]
```diff
@@ -227,6 +227,29 @@ const setInputsForCommand = (instanceYaml, command, config) => {
...
+      return;
+    }
+    if (Array.isArray(evt.data.logs)) {
+      evt.data.logs.forEach((log) => {
+        // Remove strange formatting that comes from stderr
+        if (log.data.startsWith("'")) {
+          log.data = log.data.slice(1);
+        }
+        if (log.data.endsWith("'")) {
+          log.data = log.data.slice(0, log.data.length - 1);
```
Comment (medikoo) [inline src/cli/commands-cn/utils.js:242]: nit: It can be just `log.data.slice(0, -1);`

Comment (hkbarton) [reply src/cli/commands-cn/utils.js:242]: done

**Pull request merged**